server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupProvisioning.java

```diff
@@ -324,4 +349,12 @@ protected void validateOrderBy(String orderBy) throws IllegalArgumentException {
         super.validateOrderBy(orderBy, GROUP_FIELDS);
     }
 
+    private void validateAllowedUserGroups(String zoneId, ScimGroup group) {
+        Set<String> allowedGroups = getAllowedUserGroups(zoneId);
+        if ((allowedGroups != null) && (!allowedGroups.contains(group.getDisplayName()))) {
```

Comment:
I don't like that we are doing the filtering based on the group display name. That feels quite dangerous.

Is it because in this specific case the "display name" is actually the unique string that represents it as well? If that's the case, could we then rename that method to be `group.identifier()` instead of `getDisplayName()`?

Reply:
ok, but the groups in UAA what in reality is used. UAA uses the SCIM group for authorizations and the display name is the scope name. The identitfier is a not used entity , also in other usage of it